### PR TITLE
Increase CW and EMF log event size limit from 256KB to 1MB

### PR DESCRIPTION
**Pull request by TravisStark**
# Increase CloudWatch Logs event size limit from 256KB to 1MB

## Description

This PR updates the internal/aws/cwlogs package to support the CloudWatch Logs event size limit increase from 256KB to 1MB. The changes ensure that log events up to 1MB in size can be properly processed and sent to CloudWatch Logs without truncation.

No changes made to truncation logic.

## Changes

- Increased `defaultMaxEventPayloadBytes` constant from 256KB to 1MB
- Fixed a bug in the `exceedsLimit` function by using `maxRequestPayloadBytes` instead of `maxEventPayloadBytes` for batch size checks
- Added test cases to verify the behavior with the new limit:
  - Events at exactly 1MB
  - Events over 1MB (truncation)
  - Events between 256KB and 1MB
  - Batch handling with large events
  - Correct limit usage for batch size checks

## Testing
Built into cw agent and deployed on EC2 without errors.

### Passing Stress Tests
- https://github.com/aws/amazon-cloudwatch-agent/actions/runs/16328788036
- https://github.com/aws/amazon-cloudwatch-agent/actions/runs/16327798828

Added unit tests to verify edge cases and behaviors with the new limit. Checks:
- Events up to 1MB pass through unchanged
- Events over 1MB are properly truncated
- Batch size limits are correctly enforced
- Events between the old limit (256KB) and new limit (1MB) are handled correctly

### CODE REVIEW

File: internal/aws/cwlogs/pusher.go [hunk 1]
```diff
@@ -16,8 +16,9 @@ import (
 
 const (
 	// http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html
+	// Updated to support 1MB log events
 	// In truncation logic, it assuming this constant value is larger than perEventHeaderBytes + len(truncatedSuffix)
-	defaultMaxEventPayloadBytes = 1024 * 256 // 256KB
+	defaultMaxEventPayloadBytes = 1024 * 1024 * 1 // 1MB (increased from 256KB)
```
Comment (dricross) [inline internal/aws/cwlogs/pusher.go]: nit: don't need the `Updated to support...` comment or the `increase from...` part of the comment

File: internal/aws/cwlogs/pusher.go [hunk 2]
```diff
@@ -121,7 +122,7 @@ func newEventBatch(key StreamKey) *eventBatch {
 
 func (batch eventBatch) exceedsLimit(nextByteTotal int) bool {
 	return len(batch.putLogEventsInput.LogEvents) == cap(batch.putLogEventsInput.LogEvents) ||
-		batch.byteTotal+nextByteTotal > maxEventPayloadBytes
+		batch.byteTotal+nextByteTotal > maxRequestPayloadBytes
```
Comment (dricross) [inline internal/aws/cwlogs/pusher.go]: `maxRequestPayloadBytes` seems to be an unused variable (only referenced in test code) that should probably get refactored out. I think we should stick with `maxEventPayloadBytes`

**Comment by dricross** [comment]
I can't comment on it directly, but there's a reference to the 256KB event size limit on line 228 of cwlogs/pusher.go that we should update: https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/aws-cwa-dev/internal/aws/cwlogs/pusher.go#L228

File: internal/aws/cwlogs/pusher_test.go [hunk 3]
```diff
@@ -278,3 +278,100 @@ func TestMultiStreamPusher(t *testing.T) {
 	assert.Equal(t, "foo", *inputs[1].LogGroupName)
 	assert.Equal(t, "bar2", *inputs[1].LogStreamName)
 }
+
+// Test cases for 1MB event support
+func TestEventValidation_1MBLimit(t *testing.T) {
+	logger := zap.NewNop()
+
+	// Test event at exactly 1MB minus headers and suffix
+	maxMessageSize := defaultMaxEventPayloadBytes - perEventHeaderBytes - len(truncatedSuffix)
```
Comment (dricross) [inline internal/aws/cwlogs/pusher_test.go]: Would a message at defaultMaxEventPayloadBytes - perEventHeaderBytes be truncated? I don't think it should be.

Comment (TravisStark) [reply internal/aws/cwlogs/pusher_test.go]: Good catch, removed len(truncatedSuffix) subraction

**Review by okankoAMZ — APPROVED** [review]

**Review by dricross — APPROVED** [review]

**Review by dricross — COMMENTED** [review]
We should run the stress tests to see what kind of memory impact this will have
